src/payment/payjoin.rs

```diff
@@ -0,0 +1,80 @@
+//! Holds a payment handler allowing to send Payjoin payments.
+
+use crate::types::PayjoinSender;
+use crate::{error::Error, Config};
+
+use std::sync::{Arc, RwLock};
+
+/// A payment handler allowing to send Payjoin payments.
+///
+/// Should be retrieved by calling [`Node::payjoin_payment`].
+///
+/// [`Node::payjoin_payment`]: crate::Node::payjoin_payment
+pub struct PayjoinPayment {
+	runtime: Arc<RwLock<Option<tokio::runtime::Runtime>>>,
+	sender: Arc<PayjoinSender>,
+	config: Arc<Config>,
+}
+
+impl PayjoinPayment {
+	pub(crate) fn new(
+		runtime: Arc<RwLock<Option<tokio::runtime::Runtime>>>, sender: Arc<PayjoinSender>,
+		config: Arc<Config>,
+	) -> Self {
+		Self { runtime, sender, config }
+	}
+
+	/// Send an on chain Payjoin transaction to the address specified in the `payjoin_uri`
+	/// following the spec of [`BIP77`].
+	///
+	/// The `payjoin_uri` parameter is a [`BIP21`] URI that contains information about the Payjoin
+	/// receiver.
+	///
+	/// The `amount` parameter is optional, and if not provided, the amount from the `payjoin_uri`
+	/// will be used. If neither is provided, an error will be returned. `fee_rate` is also
+	/// optional, and if not provided, the wallet's default fee rate will be used.
+	///
+	/// The method will return the transaction ID of the Payjoin transaction if it was sent and a
+	/// valid response was received. If the response is not yet available, the method will return
+	/// `None` and the response will be processed asynchronously.
+	///
+	/// [`BIP21`]: https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki
+	/// [`BIP77`]: https://github.com/bitcoin/bips/blob/d7ffad81e605e958dcf7c2ae1f4c797a8631f146/bip-0077.mediawiki
+	pub async fn send(
```

Comment:
so I need this to be async as I want to call `sender.fetch` but still might need the runtime to start a background process. is that problematic?